src/System.Slices/src/System/SpanEnumerators.cs

```diff
@@ -88,13 +96,7 @@ public void Dispose()
 
             public bool MoveNext()
             {
-                int nextItemIndex = _position + 1;
-                if (nextItemIndex < _span.Length)
-                {
-                    _position = nextItemIndex;
-                    return true;
-                }
-                return false;
+                return ++_position < _span.Length;
```

Comment:
It may be a good idea to have a testcase for IEnumerator.MoveNext being called enough times to cause overflow. It should be fairly quick on todays hardware and iterator should tolerate that.